dotcom-rendering/src/web/browser/islands/doHydration.ts

```diff
@@ -14,40 +14,45 @@ import { initPerf } from '../initPerf';
  * @param data The deserialised props we want to use for hydration
  * @param element The location on the DOM where the component to hydrate exists
  */
-export const doHydration = (name: string, data: any, element: HTMLElement) => {
+export const doHydration = async (
+	name: string,
+	data: any,
+	element: HTMLElement,
+) => {
 	// If this function has already been run for an element then don't try to
 	// run it a second time
 	const alreadyHydrated = element.dataset.guReady;
 	if (alreadyHydrated) return;
 
 	const { start, end } = initPerf(`hydrate-${name}`);
 	start();
-	import(
-		/* webpackInclude: /\.importable\.tsx$/ */
-		/* webpackChunkName: "[request]" */
-		`../../components/${name}.importable`
-	)
-		.then((module) => {
-			const clientOnly = element.getAttribute('clientOnly') === 'true';
 
-			if (clientOnly) {
-				element.querySelector('[data-name="placeholder"]')?.remove();
-				log('dotcom', `Rendering island ${name}`);
-				render(h(module[name], data), element);
-			} else {
-				log('dotcom', `Hydrating island ${name}`);
-				hydrate(h(module[name], data), element);
-			}
+	try {
+		const module = await import(
```

Comment:
We don't use this pattern anywhere else when we import. Is there a particular problem it's solving here or is this just code style preference?

Reply:
more of a question, really… I’m curious as to why we don’t.

Reply:
I think people just follow idiomatic conventions. I looked at the first 8 results from Google on how to use dynamic imports and none of them suggested `await` 🤷

Reply:
Counterpoint to this, we have 3 instances in the codebase where we use `await import`:

```
dotcom-rendering/src/web/browser/sentryLoader/init.ts:
  48  			}
  49  
  50  			// Load sentry.ts
  51: 			const { reportError } = await import(
  52  				/* webpackChunkName: "sentry" */ './sentry'
  53  			);
  54  

dotcom-rendering/src/web/components/YoutubeBlockComponent.importable.tsx:
  91  
  92  	useEffect(() => {
  93  		const defineConsentState = async () => {
  94: 			const { onConsentChange } = await import(
  95  				'@guardian/consent-management-platform'
  96  			);
  97  			onConsentChange((newConsent: ConsentState) => {

dotcom-rendering/src/web/lib/braze/initialiseAppboy.ts:
  11  };
  12  
  13  const initialiseAppboy = async (apiKey: string): Promise<typeof appboy> => {
  14: 	const importedAppboy = (await import(
  15  		/* webpackChunkName: "braze-web-sdk-core" */ '@braze/web-sdk-core'
  16  	)) as unknown as typeof appboy;
  17  
```